ydb/core/tx/schemeshard/schemeshard_utils.h

```diff
@@ -183,27 +186,48 @@ bool CommonCheck(const TTableDesc& tableDesc, const NKikimrSchemeOp::TIndexCreat
         return false;
     }
 
-    if (!indexKeys.DataColumns.empty() && !AppData()->FeatureFlags.GetEnableDataColumnForIndexTable()) {
-        status = NKikimrScheme::EStatus::StatusPreconditionFailed;
-        error = "It is not allowed to create index with data column";
-        return false;
+    if (!indexKeys.DataColumns.empty()) {
+        if (!AppData()->FeatureFlags.GetEnableDataColumnForIndexTable()) {
+            status = NKikimrScheme::EStatus::StatusPreconditionFailed;
+            error = "It is not allowed to create index with data column";
+            return false;
+        }
     }
 
-    if (!IsCompatibleIndex(baseTableColumns, indexKeys, error)) {
+    if (!IsCompatibleIndex(indexDesc.GetType(), baseTableColumns, indexKeys, error)) {
         status = NKikimrScheme::EStatus::StatusInvalidParameter;
         return false;
     }
 
-    TColumnTypes columnsTypes;
-    if (!ExtractTypes(tableDesc, columnsTypes, error)) {
+    TColumnTypes baseColumnTypes;
+    if (!ExtractTypes(tableDesc, baseColumnTypes, error)) {
         status = NKikimrScheme::EStatus::StatusInvalidParameter;
         return false;
     }
 
-    implTableColumns = CalcTableImplDescription(baseTableColumns, indexKeys);
-    if (!IsCompatibleKeyTypes(columnsTypes, implTableColumns, uniformTable, error)) {
-        status = NKikimrScheme::EStatus::StatusInvalidParameter;
-        return false;
+    implTableColumns = CalcTableImplDescription(indexDesc.GetType(), baseTableColumns, indexKeys);
+
+    if (indexDesc.GetType() == NKikimrSchemeOp::EIndexType::EIndexTypeGlobalVectorKmeansTree) {
+        if (indexKeys.KeyColumns.size() != 1) {
+            status = NKikimrScheme::EStatus::StatusInvalidParameter;
+            error = "Only single column is supported for vector index";
+            return false;
+        }
+
+        const TString& indexColumnName = indexKeys.KeyColumns[0];
+        Y_ABORT_UNLESS(baseColumnTypes.contains(indexColumnName));
+        auto typeInfo = baseColumnTypes.at(indexColumnName);
+
+        if (typeInfo.GetTypeId() != NScheme::NTypeIds::String) {
```

Comment:
Is nullable and not nullable string have same TypeIds?

Reply:
NotNull is the separate field.
[Details here.](https://github.com/ydb-platform/ydb/blob/0fa3052f7c623d25d997e4844c4f1672be40efca/ydb/core/protos/flat_scheme_op.proto#L94)